include/SDL_hints.h

```diff
@@ -2048,6 +2048,20 @@ extern "C" {
  */
 #define SDL_HINT_WINDOWS_DPI_SCALING "SDL_WINDOWS_DPI_SCALING"
 
+/**
+ * \brief Opts created windows into the Windows DWM dark mode appearance.
+ *
+ *  By default, Windows does not opt any applications into adhering to the dark
+ *  mode color scheme for title bars/borders. This hint tells Windows to use
+ *  the appropriate dark/light mode colors for created SDL windows.
+ *
+ *  This variable can be set to the following values:
+ *    "0"       - Windows will always use the light mode color scheme.
+ *    "1"       - Windows will follow the user light/dark mode preference and use
+ *                dark title bar and window borders when appropriate.
```

Comment:
Based on the name of the hint, I would expect to always get dark mode if I set this. If this is only opting in to follow the system light/dark preference, I think we should use a name that describes behavior specifically.

Reply:
Yeah, maybe something like `SDL_HINT_WINDOWS_HONOR_DARK_MODE` or something, not sure. But that's getting a bit wordy. Ideas?

Reply:
Silly question, is there a case where we _don't_ want to honor the OS preference? I'm thinking we just add dark mode support and follow the OS preference for all platforms, and if we need to add a hint, it's to disable that behavior.

Reply:
Can you please split this out into a separate PR? I'd like to merge it later once we add dark mode support (possibly in SDL3)